src/user_interface.py

```diff
@@ -0,0 +1,41 @@
+import re
+
+from src.message import Message as message
+
+
+class UserInterface:
+    def get_user_input(self):
+        user_input = input()
+        is_integer = self.determine_is_integer(user_input)
```

Comment:
Seems like checking if it's an integer is a part of validating the move. How come you check it before validating the move?

Reply:
My reasoning is that `is_integer` is just checking if the input is 1-9 (but the input comes in as a string).  Then, if it is a string-integer 1-9, we can turn it into an actual `integer` and return it.  If not, it will be handled with `handle_incorrect_input`.  Does that make sense?

Reply:
It makes sense, but checking if it's an integer seems like a validation step to me, so I'd expect it to be encapsulated in the validation logic